parsec/api/data/invite.py

```diff
@@ -13,21 +13,24 @@
 import attr
 
 
+SASCodeTypeVar = TypeVar("SASCodeTypeVar", bound="SASCode")
+
+
 class SASCode(str):
     __slots__ = ()
     length = 4
     symbols = "ABCDEFGHJKLMNPQRSTUVWXYZ23456789"
     regex = re.compile(rf"^[{symbols}]{{{length}}}$")
 
-    def __init__(self, raw):
+    def __init__(self, raw: str):
         if not isinstance(raw, str) or not self.regex.match(raw):
             raise ValueError("Invalid SAS code")
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f"<SASCode {super().__repr__()}>"
 
     @classmethod
-    def from_int(cls, num):
+    def from_int(cls: Type[SASCodeTypeVar], num: int) -> SASCodeTypeVar:
```

Comment:
Why not simply `def from_int(cls, num: int) -> "SASCode":`?

Reply:
fixed